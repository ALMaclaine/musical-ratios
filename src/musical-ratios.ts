export enum MusicalRatios {
  Unison = 1.00,
  MinorSecond = 1.067,
  MajorSecond = 1.125,
  MinorThird = 1.200,
  MajorThird = 1.250,
  PerfectFourth = 1.333,
  DiminishedFifth = 1.400,
  AugmentedFourth = 1.414,
  PerfectFifth = 1.500,
  MinorSixth = 1.600,
  GoldenRatio = 1.618,
  MajorSixth = 1.667,
  MinorSeventh = 1.750,
  MajorSeventh = 1.875,
  Octave = 2.000
}

export function intervalToRatio(interval) {
  return MusicalRatios[interval];
}

export function ratioToInterval(ratio) {
  return MusicalRatios[ratio];
}

export function ratioToPower(ratio: MusicalRatios | number, power: number) {
  return Math.pow(ratio, power);
}
