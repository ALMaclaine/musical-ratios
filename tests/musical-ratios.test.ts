import * as npmUtils from '../src';

describe('musical-ratios', () => {
    it('Has Correct API', () => {
        const keys = Object.keys(npmUtils);
        expect(keys.length).toBe(4);
        expect(keys).toMatchSnapshot();
    });
});
