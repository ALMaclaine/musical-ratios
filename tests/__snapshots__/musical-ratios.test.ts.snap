// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`musical-ratios Has Correct API 1`] = `
Array [
  "MusicalRatios",
  "intervalToRatio",
  "ratioToInterval",
  "ratioToPower",
]
`;
